packages/libs/deer/src/lib.rs

```diff
@@ -44,23 +44,61 @@ mod schema;
 extern crate alloc;
 
 pub trait ObjectAccess<'de> {
+    /// This enables bound-checking for [`ObjectAccess`]. After calling this [`ObjectAccess`] will
+    /// ensure that there are never more than `length` values returned by [`Self::next`], if there
+    /// are not enough items present [`ArrayAccess`] will call [`Visitor::visit_none`], for
+    /// [`Self::value`] calls [`Visitor::visit_none`] will be called on the tuple of `(K, V)`, while
+    /// [`Self::value`] will call [`Visitor::visit_none`] of `V`.
+    ///
+    /// [`Self::value`] also counts toward the length, behaviour of multiple calls to
+    /// [`Self::value`] will always decrement the counter.
+    ///
+    /// This is best suited for types where the length/amount of keys is already predetermined, like
+    /// structs or enum variants.
+    ///
+    /// # Errors
+    ///
+    /// This should error if a call to [`Self::next`] or [`Self::value`] has been made before
+    /// calling this function or this function has been called repeatably.
```

Comment:
I don't know if there are any conventions on documentation on traits, but I'd prefer avoiding "should". This is a hint to the implementor, but the majority of users call that function instead of implementing it.

Reply:
fixed in [df9fb46](https://github.com/hashintel/hash/pull/1645/commits/df9fb46e67cea4dfeb63eb8ef61879f2d54340dd)